### PR TITLE
Edit menu and add dishes to it

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT Story: https://www.pivotaltracker.com/story/show/131350845

Changes proposed in this pull request:
- Create edit menu page and functionality
- Add dishes to edit menu page
- Display dishes on menu page
- Dishes on menu page are links to their dish page
- Delete contents of previous menu_page.feature because they test something we don't actually want to happen in our app (displaying a static `/menu` - we will only have `/menu/:id` )

What I have learned working on this feature:
- Better knowledge about how basic Rails routing and info shuffling works

Screenshots:
screen shot 2016-10-02 at 9 25 37 pm
screen shot 2016-10-02 at 9 25 51 pm

Ready for review!

### CODE REVIEW

File: app/controllers/menus_controller.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@ def new
     @menu = Menu.new
   end
 
+  def show
+    @menu = Menu.find(params[:id])
```
Comment (tochman) [inline app/controllers/menus_controller.rb]: Repeated in several methods. Plz extract to a private method and set as before action

File: app/controllers/menus_controller.rb [hunk 2]
```diff
@@ -17,8 +21,24 @@ def create
...
+  end
+
+  def update
+    @menu = Menu.find(params[:id])
+    @menu.update(menu_params)
+    # params[:menu][:dish_ids].each do |dish|
+    #   @menu.dish_ids << dish
+    #   @menu.save
+    # end
+    # binding.pry
```
Comment (tochman) [inline app/controllers/menus_controller.rb]: Comments...

File: features/edit_menu.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    | title |
+    | Lunch |
+  And I am on the edit menu page for "Lunch"
+  Then I should see:
+    | content    |
+    | Title      |
+    | add dishes |
+    | Pizza      |
+    | Salad      |
+    | Olives     |
```
Comment (tochman) [inline features/edit_menu.feature]: I would skip the "I should see..." steps

File: features/edit_menu.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Scenario: I edit my menu
+  Given the following dishes exist
+    | dish_name | dish_desc       | dish_price |
+    | Pizza     | Delicious pizza | 7000       |
+    | Salad     | Leafy           | 1500       |
+    | Olives    | Salty           | 900        |
+  And the following menus exist:
+    | title |
+    | Lunch |
```
Comment (tochman) [inline features/edit_menu.feature:13]: All of the above could be placed in a background for better readability

File: features/step_definitions/menu_steps.rb [hunk 5]
```diff
@@ -16,3 +16,22 @@
...
+
+Given(/^I am on the edit menu page for "([^"]*)"$/) do |menu_name|
+  @menu = Menu.find_by(title: menu_name)
+  visit edit_menu_path(@menu)
+end
+
+Given(/^I add a dish to the "([^"]*)" menu$/) do |menu|
+  menu_page = Menu.find_by(title: menu)
+  visit(edit_menu_path(menu_page))
+  steps %Q{
```
Comment (tochman) [inline features/step_definitions/menu_steps.rb:28]: Small q if you don't do any interpolation. On the other hand perhaps you should do just that. Hard coded values (like Pizza in this case) are usually not desirable.

**Review by tochman — CHANGES_REQUESTED** [review]

File: app/controllers/menus_controller.rb (hunk 1 above)
Comment (AmberWilkie) [reply app/controllers/menus_controller.rb]: done

File: app/controllers/menus_controller.rb (hunk 2 above)
Comment (AmberWilkie) [reply app/controllers/menus_controller.rb]: Sorry about that, was rushing to push this out before George walked in the door.

File: features/edit_menu.feature (hunk 3 above)
Comment (AmberWilkie) [reply features/edit_menu.feature]: ok I got rid of them. I see how we're testing for (most of) them later.

File: features/step_definitions/menu_steps.rb (hunk 5 above)
Comment (AmberWilkie) [reply features/step_definitions/menu_steps.rb:28]: Refactored. I know better, that's sloppy.

**Comment by tochman** [comment]
Good work. I would like you to revisit the UX in a separate feature but it works for now.

**Pull request merged**